### DISCUSSION TITLE
A way of inserting http header in sse stream response?

### DISCUSSION THREAD

**Discussion by loldruger** [tokio-rs/axum#2214] [Q&A] [answered]
i have a problem that sse stream response has been buffered by nginx so 
```rust
    let stream = stream::repeat_with(|| Event::default().data("hi!"))
        .take(5)
        .map(Ok)
        .throttle(Duration::from_secs(1));

    let sse = Sse::new(stream)
        .keep_alive(KeepAlive::default());
```
this code gets buffered and no response to client till 5 seconds goes. 
the solution i found is to get `x-accel-buffering` header off! 

https://serverfault.com/questions/801628/for-server-sent-events-sse-what-nginx-proxy-configuration-is-appropriate

so im finding a way of inserting http header in sse stream response. but it seems not possible currently. 
is there any opinion or solution do you know? without touching nginx configuration.

**Comment by davidpdrsn** [comment]
See https://docs.rs/axum/0.6.20/axum/response/index.html#building-responses

**Reply by loldruger** [reply]
it seems to have to implement stream response manually.. but how? the link has no example with this

**Reply by davidpdrsn** [reply, marked as answer]
No, you can return `(header_map, sse)`
